internal/engine/wazevo/backend/regalloc/reg.go

```diff
@@ -96,6 +213,7 @@ const (
 	RegTypeInt
 	RegTypeFloat
 	RegTypeNum
+	numRegTypes // keep last
```

Comment:
I think we can use `RegTypeNum` for numRegTypes :D 
```suggestion
	RegTypeNum
```

Reply:
Thanks for pointing out that the constant already existed. I got confused at first and assumed `RegTypeNum` was a type of registry (like _number_ 🤷) so I took the liberty to rename the constant to `NumRegType` to differentiate it and make it more clear that it's not representing a registry type. I can revert if you prefer the other spelling; let me know!